packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,258 @@
+import { glob } from 'glob';
+import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import {
+  CloudFormationClient,
+  DeleteStackCommand,
+} from '@aws-sdk/client-cloudformation';
+import { BackendIdentifierConversions } from '@aws-amplify/platform-core';
+import { getClientConfigPath } from '@aws-amplify/client-config';
+import { TestBranch, amplifyAppPool } from '../amplify_app_pool.js';
+import { e2eToolingClientConfig } from '../e2e_tooling_client_config.js';
+
+type PackageManager = 'npm' | 'yarn-classic' | 'yarn-modern' | 'pnpm';
+type PackageManagerExecutable = 'npx' | 'yarn' | 'pnpm';
+
+const setupPackageManager = async (
+  dir: string
+): Promise<{
+  packageManager: PackageManager;
+  packageManagerExecutable: PackageManagerExecutable;
+}> => {
+  const packageManager = (process.env.PACKAGE_MANAGER_EXECUTABLE ??
+    'npm') as PackageManager;
+  const packageManagerExecutable = packageManager.startsWith('yarn')
+    ? 'yarn'
+    : packageManager === 'npm'
+    ? 'npx'
+    : (packageManager as PackageManagerExecutable);
+  const execaOptions = {
+    cwd: packageManager === 'yarn-modern' ? dir : os.homedir(),
+    stdio: 'inherit' as const,
+  };
+
+  if (packageManager === 'npm') {
+    // nuke the npx cache to ensure we are installing packages from the npm proxy
+    const { stdout } = await execa('npm', ['config', 'get', 'cache']);
+    const npxCacheLocation = path.join(stdout.toString().trim(), '_npx');
+
+    if (existsSync(npxCacheLocation)) {
+      await fsp.rm(npxCacheLocation, { recursive: true });
+    }
+  } else if (packageManager.startsWith('yarn')) {
+    if (packageManager === 'yarn-modern') {
+      await execa('corepack', ['enable'], execaOptions);
+      await execa('yarn', ['init', '-2'], execaOptions);
+
+      await execa(
+        'yarn',
+        ['config', 'set', 'npmRegistryServer', 'http://localhost:4873'],
+        execaOptions
+      );
+      await execa(
+        'yarn',
+        ['config', 'set', 'unsafeHttpWhitelist', 'localhost'],
+        execaOptions
+      );
+      await execa(
+        'yarn',
+        ['config', 'set', 'nodeLinker', 'node-modules'],
+        execaOptions
+      );
+    } else {
+      await execa('npm', ['install', '-g', 'yarn'], { stdio: 'inherit' });
+      await execa(
+        'yarn',
+        ['config', 'set', 'registry', 'http://localhost:4873'],
+        execaOptions
+      );
+      await execa('yarn', ['config', 'get', 'registry'], execaOptions);
+    }
+    await execa('yarn', ['cache', 'clean'], execaOptions);
+  } else if (packageManager === 'pnpm') {
+    await execa('npm', ['install', '-g', packageManager], {
+      stdio: 'inherit',
+    });
+    await execa(packageManager, ['--version']);
+    await execa(packageManager, [
+      'config',
+      'set',
+      'registry',
+      'http://localhost:4873',
+    ]);
+    await execa(packageManager, ['config', 'get', 'registry']);
+  }
+
+  return { packageManagerExecutable, packageManager };
+};
```

Comment:
Couple of things:
1. given the volume I suggest to extract this to separate file.
2. it seems that `PACKAGE_MANAGER_EXECUTABLE` env var should really be `PACKAGE_MANAGER` 
3. this code seems to look like some sort of mapping, i.e. it maps `PACKAGE_MANAGER_EXECUTABLE/PACKAGE_MANAGER` env var into certain actions that are relevant for given package manager. and the way this code will likely evolve is that we'll be adding more varieties. So instead of scattering if/else conditions around I'd just group logice per package manager kind. Something like this:

```
export const setupPackageManager = async (...) => {
   const packageManager = (process.env.PACKAGE_MANAGER_EXECUTABLE ??
    'npm') as PackageManager;
   switch(packageManager) {
      case 'npm':
          packageManagerExecutable = 'npx';
          await initializeNpm();
          break;
      case 'yarn-modern:
          packageManagerExecutable = 'yarn';
          await initializeYarnModern();
          break;
      case 'yarn-classic:
          packageManagerExecutable = 'yarn';
          await initializeYarnClassic();
          break;
   }
  
}

```

This will make future editors of this function understand better what they have to do to add support for another package manager.

Reply:
Done 0eb8919bb7